flow/Deque.h

```diff
@@ -168,10 +168,18 @@ class Deque {
 		if (newSize > max_size()) throw std::bad_alloc();
 		//printf("Growing to %lld (%u-%u mask %u)\n", (long long)newSize, begin, end, mask);
 		T* newArr = (T*)aligned_alloc(std::max(__alignof(T), sizeof(void*)),
-		                              newSize * sizeof(T)); // SOMEDAY: FastAllocator, exception safety
+		                              newSize * sizeof(T)); // SOMEDAY: FastAllocator
 		ASSERT(newArr != nullptr);
+		try {
+			for (int i = begin; i != end; i++) {
+				new (&newArr[i - begin]) T(std::move_if_noexcept(arr[i & mask]));
+			}
+		} catch (...) {
+			aligned_free(newArr);
```

Comment:
I think we need to destroy each object that was successfully constructed here too, otherwise we could have a leak.